### PR TITLE
Feature/add secrets to mo services

### PR DESCRIPTION
**Pull request by valentin-mogenius**

### CODE REVIEW

File: dtos/k8sEsoSettingsDto.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package dtos
+
+type K8sEsoSettingsDto struct {
+	SecretStoreNamePrefix string `json:"secretStoreNamePrefix" validate:"required"`
```
Comment (valentin-mogenius) [inline dtos/k8sEsoSettingsDto.go]: take out required.

Comment (beneiltis) [reply dtos/k8sEsoSettingsDto.go]: done

File: kubernetes/deployment.go [hunk 2]
```diff
@@ -89,29 +90,37 @@ func DeleteDeployment(job *structs.Job, namespace dtos.K8sNamespaceDto, service
 	}(wg)
 }
 
+func GetDeployment(namespace, deploymentName string) (*v1.Deployment, error) {
```
Comment (beneiltis) [inline kubernetes/deployment.go]: removed. was already in punq.

File: kubernetes/secret.go [hunk 3]
```diff
@@ -412,3 +404,8 @@ func watchSecrets(provider *punq.KubeProvider, kindName string) error {
 		}
 	}
 }
+
+func ListSecrets(namespace string) (*v1.SecretList, error) {
+	client := getCoreClient()
```
Comment (beneiltis) [inline kubernetes/secret.go]: removed. also available in punq.

File: kubernetes/service.go [hunk 4]
```diff
@@ -129,18 +95,23 @@ func UpdateService(job *structs.Job, namespace dtos.K8sNamespaceDto, service dto
...
-// 		return err
-// 	}
-// 	serviceClient := provider.ClientSet.CoreV1().Services("")
-// 	_, err = serviceClient.Update(context.TODO(), service, metav1.UpdateOptions{})
-// 	if err != nil {
-// 		return err
-// 	}
-// 	return nil
-// }
+func GetService(namespace, serviceName string) (*v1Core.Service, error) {
```
Comment (beneiltis) [inline kubernetes/service.go]: removed. available in punq.

File: kubernetes/service.go [hunk 5]
```diff
@@ -129,18 +95,23 @@ func UpdateService(job *structs.Job, namespace dtos.K8sNamespaceDto, service dto
...
+	client := getCoreClient().Services(namespace)
+	return client.Get(context.TODO(), serviceName, metav1.GetOptions{})
+}
+
+func UpdateServiceWith(service *v1.Service) error {
```
Comment (beneiltis) [inline kubernetes/service.go]: removed. avaliable in punq and unused.

**Pull request merged**